sim/core/character.go

```diff
@@ -165,8 +170,27 @@ func (character *Character) AddStats(stat stats.Stats) {
 	character.stats = character.stats.Add(stat)
 }
 func (character *Character) AddStat(stat stats.Stat, amount float64) {
+	if stat == stats.MeleeHaste {
```

Comment:
I think it should be fine to put the check in here and delete AddMeleeHaste(). This isn't really on the hot path so I don't think the performance difference is that high? Either way I'd rather do it the simple way first, and optimize later.

Reply:
The function signature has to change so I made a new function